src/cairoUtilFuncGen/calldata/implicitArrayConversion.ts

```diff
@@ -0,0 +1,503 @@
+
+  private DynamicToDynamicConversion(
+    key: string,
+    targetType: TypeNode,
+    sourceType: TypeNode,
+  ): CairoFunction {
+    assert(targetType instanceof PointerType && sourceType instanceof PointerType);
+    assert(targetType.to instanceof ArrayType && sourceType.to instanceof ArrayType);
+
+    assert(targetType.to.size === undefined && sourceType.to.size === undefined);
+
+    const targetElmType = targetType.to.elementT;
+    const sourceElmType = sourceType.to.elementT;
+
+    const funcName = `CD_DY_TO_WS_DY${this.generatedFunctions.size}`;
+    this.generatedFunctions.set(key, { name: funcName, code: '' });
+
+    const cairoSourceElementType = CairoType.fromSol(
+      sourceType.to.elementT,
+      this.ast,
+      TypeConversionContext.CallDataRef,
+    );
+
+    const cairoTargetElementType = CairoType.fromSol(
+      targetType.to.elementT,
+      this.ast,
+      TypeConversionContext.StorageAllocation,
+    );
+
+    const cairoSourceType = CairoType.fromSol(
+      sourceType,
+      this.ast,
+      TypeConversionContext.CallDataRef,
+    );
+    assert(cairoSourceType instanceof CairoDynArray);
+    const cairoSourceTypeString = cairoSourceType.toString();
+    const dynArrayLengthName = this.dynArrayGen.gen(cairoTargetElementType)[1];
+    const implicit =
+      '{syscall_ptr : felt*, range_check_ptr, pedersen_ptr : HashBuiltin*, bitwise_ptr : BitwiseBuiltin*}';
+    const loaderName = `DY_LOADER${this.generatedFunctions.size}`;
+    let code;
+    if (sourceElmType instanceof IntType && targetElmType instanceof IntType) {
+      code = [
+        `func ${loaderName}${implicit}(ref: felt, len: felt, ptr: ${cairoSourceType.ptr_member.toString()}*, target_index: felt):`,
+        `    alloc_locals`,
+        `    if len == 0:`,
+        `      return ()`,
+        `    end`,
+        `    let (storage_loc) = ${this.dynArrayIndexAccessGen.getOrCreate(
+          targetElmType,
+        )}(ref, Uint256(target_index, 0))`,
+        `    let target_index = target_index + 1`,
+        sourceElmType.signed
+          ? `    let (val) = warp_int${sourceElmType.nBits}_to_int${targetElmType.nBits}(ptr[0])`
+          : `    let (val) = felt_to_uint256(ptr[0])`,
+        `    ${this.storageWriteGen.getOrCreate(targetElmType)}(storage_loc, val)`,
+        `    return ${loaderName}(ref, len - 1, ptr + ${cairoSourceElementType.width}, target_index)`,
+        `end`,
+        ``,
+
+        `func ${funcName}${implicit}(ref: felt, source: ${cairoSourceTypeString}):`,
+        `     alloc_locals`,
+        `    ${dynArrayLengthName}.write(ref, Uint256(source.len, 0))`,
+        `    ${loaderName}(ref, source.len, source.ptr, 0)`,
+        '    return ()',
+        'end',
+      ].join('\n');
+      this.requireImport(
+        'warplib.maths.int_conversions',
+        `warp_int${sourceElmType.nBits}_to_int${targetElmType.nBits}`,
+      );
+      this.requireImport('warplib.maths.utils', 'felt_to_uint256');
```

Comment:
Perhaps importing only one of this depending if you're treating signed or unsigned integers

Reply:
Yeah good idea. I am thinking of a way to compress the code, but there are such subtle differences in the actual copy code :(

Reply:
I had a similar issue in memory to storage conversion, in my case conversion code between the target and the source element type remained the same across all conversion functions so I wrapped it in a function. Perhaps the same thing can be applied here, also it will make it easier later if it's needed to add new conversion cases  like fixed bytes.